src/map/ai/controllers/pet_controller.cpp

```diff
@@ -40,7 +40,7 @@ void CPetController::Tick(time_point tick)
     TracyZoneScoped;
     TracyZoneIString(PPet->GetName());
 
-    if (PPet->isCharmed && tick > PPet->charmTime)
+    if (PPet->shouldDespawn(tick))
     {
         petutils::DespawnPet(PPet->PMaster);
```

Comment:
I didnt want to do this refactor together with this PR, but Despawnpet just calls through DetachPet always. We should remove one of the methods